Add force binary

Kotty666 opened the pull request:
needed if we've to deploy the whole gluster server in one Run.

<!--
Thank you for contributing to this project!

- This project has a Contributor Code of Conduct: https://voxpupuli.org/coc/
- Please check that here is no existing issue or PR that addresses your problem.
- Our vulnerabilities reporting process is at https://voxpupuli.org/security/

-->
#### Pull Request (PR) description
We needed the posibility to deploy the glusterfs-server at the first run, this won't work actually. So I've added the variable to force the binary path and disables the getvar() for the first run. With this we are able to add the Peers and Volume at the first run.

#### This Pull Request (PR) fixes the following issues
n/a

Kotty666 commented:
@bastelfreak are you able to Aprove the Workflow?

bastelfreak commented on manifests/peer.pp:
```diff
@@ -37,11 +37,21 @@
 define gluster::peer (
   $pool = 'default',
   $fqdn = $facts['networking']['fqdn'],
+  Optional[Boolean] $force_binary = false,
```
when it's false, you can't pass undef. So Optional doesn't make any sense. Can we just use this?
```suggestion
  Boolean $force_binary = false,
```

bastelfreak commented on manifests/volume.pp:
```diff
@@ -55,6 +55,7 @@
   Optional[Integer] $stripe                   = undef,
   Optional[Integer] $replica                  = undef,
   Optional[Integer] $arbiter                  = undef,
+  Optional[Boolean] $force_binary              = false,
```
same here
```suggestion
  Boolean $force_binary              = false,
```

Kotty666 replied:
@bastelfreak 
Without Optional is also okay, should i update it at my repo?

bastelfreak replied:
yes please. You can update your code on your own or merge the suggestion

Kotty666 replied:
@bastelfreak optional parameter have been removed.

Kotty666 commented:
@bastelfreak i've reviewed the spec test errors and fixed them. Also i removed CentOS6 from the metadata.json becaus the spec test will always fail, CentOS 6 is EOL since 4 Years and the Maintancen Updates also have been remove at Nov 2020.

bastelfreak commented on manifests/peer.pp:
```diff
@@ -35,13 +35,23 @@
   # gluster_binary fact is absent!
-  if getvar('::gluster_binary') {
+  if($force_binary) {
+    $real_binary = getvar('::gluster_binary') ? {
```
what's the purpose of this workaround? The fact we have in https://github.com/voxpupuli/puppet-gluster/blob/master/lib/facter/gluster.rb checks if the binary gluster exists in $PATH.

Kotty666 replied:
This getvar() stuff do not work at the first puppet run. So if the Server should be deployed with only 1 puppet agent run the glusterfs volume and peers won't be createt. With this workaround the binary variable will be set to the correct value so at the first run the peers an volumes could be created.

bastelfreak replied:
while this approach might works, it's really tricky to read and I think we don't use a manual `lookup()` + hiera in modules just to get a single value. Instead of all this code we could add another parameter like this:
```puppet
Stdlib::Absolutepath $gluster_binary = pick($facts['gluster_binary'], '/usr/sbin/gluster')
```

Kotty666 replied:
I will try the Stdlib version in my Test ENV.
I'll added the hiera stuff as a beginning to start removing the cases stuff from the pramas classes. But this i  would add in extra branches/PRs

Kotty666 commented:
@bastelfreak i've closed here and added a new PR (#228) in that there is a full hiera tree without params.pp and also added RHEL8 support.
